platform/ios/src/MGLMapView.h

```diff
@@ -927,21 +927,19 @@ MGL_EXPORT IB_DESIGNABLE
 - (MGLMapCamera *)cameraThatFitsCoordinateBounds:(MGLCoordinateBounds)bounds edgePadding:(UIEdgeInsets)insets;
 
 /**
- Returns the camera that best fits the given coordinate bounds, optionally with
- some additional padding on each side.
+ Returns the camera that best fits the given shape, with the specified camera,
+ optionally with some additional padding on each side.
  
- @param bounds The coordinate bounds to fit to the receiver’s viewport.
- @param direction The direction of the viewport, measured in degrees clockwise from true north.
- @param pitch The viewing angle of the camera, measured in degrees. A value of
-    `0` results in a camera pointed straight down at the map. Angles greater
-    than `0` result in a camera angled toward the horizon.
+ @param camera The camera that the return camera should adhere to. All values
+    on this camera will be manipulated except for pitch and direction
+ @param shape The shape to fit to the receiver’s viewport.
  @param insets The minimum padding (in screen points) that would be visible
     around the returned camera object if it were set as the receiver’s camera.
- @return A camera object centered on the same location as the coordinate bounds
-    with zoom level as high (close to the ground) as possible while still
-    including the entire coordinate bounds.
+ @return A camera object centered on the shape's center with zoom level as high
+    (close to the ground) as possible while still including the entire shape. The
+    initial camera's pitch and direction will be honored.
  */
-- (MGLMapCamera *)cameraThatFitsCoordinateBounds:(MGLCoordinateBounds)bounds direction:(CLLocationDirection)direction pitch:(CGFloat)pitch edgePadding:(UIEdgeInsets)insets;
+- (MGLMapCamera *)camera:(MGLMapCamera *)camera fittingShape:(MGLShape *)shape edgePadding:(UIEdgeInsets)insets;
```

Comment:
Is there a need for a `-camera:fittingCoordinateBounds:edgePadding:` as well?

Reply:
Sure, will add.